### PR TITLE
MAINT: replace imgmath with mathjax for docs

### LINKED ISSUE #18828: Switch from imgmath to mathjax
For html doc builds we currently need to have LaTeX installed, which in general is a pain. In the Gitpod setup this also bloats the Docker image. `sphinx.ext.imgmath` needs LaTeX, `sphinx.ext.mathjax` does not.

Older discussion in gh-11753. And from https://github.com/numpy/numpy/pull/18822#issuecomment-824331826:

>> It was needed for `sphinx.ext.imgmath` to work without warnings. From my understanding, the `:math:` directives would not display anything without latex + dvisvgm.
>> 
>> (scipy uses `sphinx.ext.mathjax` so it does not run into this issue)

> Ah thanks, that makes perfect sense now. The SciPy setup works fine, and I think is actually preferable given that we can avoid LaTeX:

- discussion on moving from `mathjax` to `imgmath`: https://github.com/scipy/scipy/pull/7358
- PR to bundle `mathjax` instead: https://github.com/scipy/scipy/pull/7376
- Pandas and scikit-learn use `mathjax` as well, but have limitations for viewing rendered math offline.

I think the SciPy approach is preferred here.

_Proposal: let's switch to `mathjax` and use https://github.com/scipy/scipy-mathjax as a submodule._

### PR DESCRIPTION
**Pull request by melissawm**
This uses the same ideas as [SciPy#7376](https://github.com/scipy/scipy/pull/7376) and adds https://github.com/scipy/scipy-mathjax as a submodule.

Not sure if any other actions are needed, so I'll test and see if the CI is happy.

Closes #18828

### CODE REVIEW

**Comment by melissawm** [comment]
We can see that mathematical formulas render correctly, for example in the rendered artifact from CircleCI [here, in the "Use of internal buffers" section](https://19944-908607-gh.circle-artifacts.com/0/doc/build/html/reference/ufuncs.html#use-of-internal-buffers). However I'm not sure if some extra configuration in CI might be needed.

File: doc/source/conf.py [hunk 1]
```diff
@@ -83,11 +83,8 @@ class PyTypeObject(ctypes.Structure):
     'matplotlib.sphinxext.plot_directive',
     'IPython.sphinxext.ipython_console_highlighting',
     'IPython.sphinxext.ipython_directive',
-    'sphinx.ext.imgmath',
```
Comment (rgommers) [inline doc/source/conf.py:86]: For some reason this works, but I don't know if that's an accident or not. Why isn't `sphinx.ext.mathjax` added here?

**Review by rgommers — COMMENTED** [review]
Thanks @melissawm! Overall LGTM, result of reviewing the docs output at https://github.com/numpy/numpy/issues/18828#issuecomment-846266016.

One question about the missing Sphinx extension.

File: doc/source/conf.py (hunk 1 above)
Comment (charris) [reply doc/source/conf.py:86]: Maybe some markup is automatically rendered if you have mathjax installed on your browser? I can't see anything about `sphinx.ext.mathjax` being default in any sphinx version, I think we should use it.

Comment (charris) [reply doc/source/conf.py:86]: I also think the submodule is needed for offline use. I note the scipy module is 4 years old and mathjax has been recently updated. I'm going to push this off to 1.22 until we understand what is going on. We can backport it if needed.

Comment (charris) [reply doc/source/conf.py:86]: One thing we might want to check is browser compatibility for MathJax 3.* if we use it.
 > In particular, to allow MathJax version 3 to work with IE11, include the line
`<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>`

**Comment by melissawm** [comment]
Thanks @rgommers and @charris - @rossbar also brought up how the scipy/mathjax submodule might not even be needed. So I'll mark this as a draft for now and investigate further. I'll let you know when I have made progress on this.

**Comment by rgommers** [comment]
Why not just add the `sphinx.ext.mathjax` extension, it's a one-line change? If it works we can merge it, since the exact same patch has worked for SciPy for quite a while.

> @rossbar also brought up how the scipy/mathjax submodule might not even be needed

We had to add that because the MathJAX CDN went away.

**Comment by melissawm** [comment]
> Why not just add the `sphinx.ext.mathjax` extension, it's a one-line change? If it works we can merge it, since the exact same patch has worked for SciPy for quite a while.

Ok - let's do this and if we find a better solution in the future we can change it.

**Review by rgommers — APPROVED** [review]
Thanks Melissa!

File: doc/source/conf.py (hunk 1 above)
Comment (rgommers) [reply doc/source/conf.py:86]: > We can backport it if needed.

I don't think that is needed. The most important benefit here is to get rid of the LaTeX requirement for local doc builds and in Gitpod - both of which are not quite relevant on a release branch.

Comment (rgommers) [reply doc/source/conf.py:86]: Updating MathJAX to a more recent release could maybe give some rendering performance gains. There's no bugs or other issues in the current version AFAIK.

Comment (rgommers) [reply doc/source/conf.py:86]: We don't use any of the options except for `mathjax_path`, which points to the `scipy-mathjax` submodule. So I think there's nothing Sphinx can mess up for us here.

**Pull request merged**